packages/coordinator/src/services/translation/response/agent-person.ts

```diff
@@ -0,0 +1,149 @@
+import {fhir, hl7V3} from "@models"
+import {createPractitioner} from "./practitioner"
+import {createLocations, createOrganization, getOrganizationCodeIdentifier} from "./organization"
+import {createPractitionerRole, createRefactoredPractitionerRole} from "./practitioner-role"
+import {createPractitionerOrRoleIdentifier} from "./identifiers"
+import {prescriptionRefactorEnabled} from "../../../utils/feature-flags"
+import {addIdentifierToPractitionerOrRole} from "./common"
+
+interface TranslatedAgentPerson {
+  practitionerRole: fhir.PractitionerRole
+  practitioner?: fhir.Practitioner
+  healthcareService?: fhir.HealthcareService
+  locations: Array<fhir.Location>
+  organization?: fhir.Organization
+}
+
+class TranslatedAgentPersonFactory {
+  private readonly agentPerson: hl7V3.AgentPerson
+  private readonly prescriptionType?: string
+  private readonly representedOrganization: hl7V3.Organization
+
+  constructor(agentPerson: hl7V3.AgentPerson, prescriptionType?: string) {
```

Comment:
Hi Tom, looking at the contructor the prescription type can be null. Is there a scenerio where we wouldnt have a prescription type?

Reply:
Not that I'm aware of, there should always be either a prescription type for primary or secondary care.